File: doc-examples/nitrite-example-java/src/main/java/example/Student.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Relation;

import java.util.HashSet;
import java.util.Set;

// tag::student[]
@MappedEntity
public class Student {
    @Id
    @GeneratedValue
    private String id;

    private String name;

    @Relation(value = Relation.Kind.MANY_TO_MANY, mappedBy = "students") // <1>
    private Set<Book> books = new HashSet<>();

    public Student() {
    }

    public Student(String name) {
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Set<Book> getBooks() {
        return books;
    }

    public void setBooks(Set<Book> books) {
        this.books = books;
    }
}
// end::student[]

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/Student.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Relation
import java.util.HashSet

// tag::student[]
@MappedEntity
class Student {
    @Id
    @GeneratedValue
    var id: String? = null

    var name: String = ""

    @Relation(value = Relation.Kind.MANY_TO_MANY, mappedBy = "students")
    var books: MutableSet<Book> = HashSet() // <1>

    constructor()

    constructor(name: String) {
        this.name = name
    }
}
// end::student[]
